adapter/internal/discovery/xds/rate_limiter_cache.go

```diff
@@ -357,6 +357,24 @@ func AddSubscriptionLevelRateLimitPolicy(policyList *types.SubscriptionPolicyLis
 			rlsPolicyCache.metadataBasedPolicies[subscriptionPolicyType][policy.Organization] = make(map[string]*rls_config.RateLimitDescriptor)
 		}
 		rlsPolicyCache.metadataBasedPolicies[subscriptionPolicyType][policy.Organization][policy.Name] = descriptor
+
+		if policy.RateLimitCount > 0 && policy.RateLimitTimeUnit != "" {
+			burstCtrlUnit, err := parseRateLimitUnitFromSubscriptionPolicy(policy.RateLimitTimeUnit)
+			if err != nil {
+				loggers.LoggerXds.Error("Error while getting the burst control time unit", err)
+			}
+			burstCtrlPolicyConfig := rls_config.RateLimitPolicy{
+				Unit:            burstCtrlUnit,
+				RequestsPerUnit: uint32(policy.RateLimitCount),
+			}
```

Comment:
Does burst controlling required or optional when creating a Subscription Policy via the Admin Rest API of Choreo APIM?

Reply:
From UX level, its optional. But default value should be Unlimited. -1 / min. Do those values reach here?

Reply:
This has resolved with the if condition.

```go
if policy.RateLimitCount > 0 && policy.RateLimitTimeUnit != ""
```